creator-node/src/fileManager.js

```diff
@@ -102,50 +104,98 @@ async function saveFileToIPFSFromFS (req, srcPath, fileType, sourceFile, transac
  *  - Write file to disk.
  *  - Add file to local inode if not already.
  */
-async function saveFileForMultihash (req, multihash, expectedStoragePath) {
+async function saveFileForMultihash (req, multihash, expectedStoragePath, gatewaysToTry) {
   // If file already stored on disk, return immediately.
   if (fs.existsSync(expectedStoragePath)) {
     req.logger.info(`File already stored at ${expectedStoragePath} for ${multihash}`)
     return expectedStoragePath
   }
 
+  const storagePath = req.app.get('storagePath')
+  const filePath = expectedStoragePath.replace(storagePath, '')
+
+  // Check if the file we are trying to copy is in a directory
+  // and if so, create the directory first
+  // E.g if the path is /file_storage/QmABC/Qm123, we check if
+  // /QmABC/Qm123 contains more than 2 parts and if so, it's a directory
+  if (filePath.split('/').length > 2) {
+    const dir = path.dirname(expectedStoragePath)
```

Comment:
what if the dir already exists?

Reply:
tried this on an existing directory and it doesn't override, but it will throw an error if it exists. i'll wrap in a try-catch

Reply:
good catch, tested this out and have a fix. doesn't override the existing dir but will throw an error if it exists so wrapped in try/catch

Reply:
i think an existsSync is probably a better move than intentionally failing?

Reply:
and if we want to fail, there should be a log msg

Reply:
just gonna remove this cause it's not necessary for this pr